crates/frontend/src/pages/custom_types.rs

```diff
@@ -0,0 +1,200 @@
+use crate::components::type_template_form::utils::delete_type;
+use crate::components::{
+    drawer::{close_drawer, open_drawer, Drawer, DrawerBtn},
+    skeleton::Skeleton,
+    stat::Stat,
+    table::Table,
+    type_template_form::TypeTemplateForm,
+};
+use crate::{api::fetch_types, components::table::types::Column};
+use leptos::*;
+use serde::Deserialize;
+use serde_json::{json, Map, Value};
+
+#[derive(Debug, Clone, PartialEq)]
+struct TypeFilter {
+    pub page: i64,
+    pub count: i64,
+}
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+struct CustomTypeRow {
+    pub type_name: String,
+    pub type_schema: Value,
+}
+
+const TYPE_DRAWER_ID: &str = "type_template_drawer";
+
+#[component]
+pub fn types_page() -> impl IntoView {
+    let tenant_rs = use_context::<ReadSignal<String>>().unwrap();
+    let (filters_rs, _) = create_signal(TypeFilter { page: 1, count: 10 });
+    let types_resource = create_blocking_resource(
+        move || (tenant_rs.get(), filters_rs.get()),
+        |(t, filter)| async move {
+            match fetch_types(t, filter.page, filter.count).await {
+                Ok(types) => types.data,
+                Err(err) => {
+                    logging::log!("failed to get types due to: {:?}", err);
+                    vec![]
+                }
+            }
+        },
+    );
+    let selected_type = create_rw_signal::<Option<CustomTypeRow>>(None);
+    let table_columns = create_memo(move |_| {
+        vec![
+            Column::default("type_name".to_string()),
+            Column::default("type_schema".to_string()),
+            Column::default("created_by".to_string()),
+            Column::default("created_at".to_string()),
+            Column::default("last_modified".to_string()),
+            Column::new(
```

Comment:
Can we separate delete and edit columns?

Reply:
There are actions you can perform on a row, why separate? Also, is there a benefit to two columns?

Reply:
This is a call to be made, if we want it to be this way, will have to change it in other places(where it is two different columns). If not then change it here.

Reply:
We're doing this in overrides screen, where clone, edit and delete override are grouped for the card

Reply:
Then better to do it for the default_config page too.

Reply:
<img width="889" alt="Screenshot 2024-06-07 at 12 37 02 PM" src="https://github.com/juspay/superposition/assets/79959102/8f654551-2bde-40ee-9b18-2db5b160dfdd">
Here one can create key without choosing any schema also, is this as expected? You can show the schema field after choosing the schema.

Reply:
Yeah this is expected, the type template is optional. You could use jsonschema that hasn't been defined before.

Reply:
Maybe we can display that in any way there.